src/main/scala/com/microsoft/hyperspace/actions/RefreshDeleteAction.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.actions
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.functions._
+
+import com.microsoft.hyperspace.HyperspaceException
+import com.microsoft.hyperspace.index._
+import com.microsoft.hyperspace.index.DataFrameWriterExtensions.Bucketizer
+import com.microsoft.hyperspace.util.ResolverUtils
+
+/**
+ * Refresh index by removing index entries for deleted source data files.
+ * If some original data file(s) are removed, in order to update index,
+ * user can trigger an index refresh action during which:
+ * 1. Deleted source data files are detected;
+ * 2. Index records' lineage is leveraged to remove any index entry coming from deleted files.
+ *
```

Comment:
Can we add a note that this doesn't handle any "new" source data? (and update PR description as well?)

Reply:
updated both.

Reply:
Just so I understand this right, this is the plan, right?

  1. This PR handles the case where some files got removed from the underlying source
  2. @apoorvedave1 PR will handle the case where some files got added into the underlying source
  3. #149 will take care of merging the logic

Reply:
nit: Can you also link to the issue that will handle this addition of "new" source data? Once we address #149, we can get rid of that line.

Reply:
@rapoth - Issue #149 is already mentioned in the first line of PR description. Just to be clear, do you want apoordave1 PR to be mentioned in this PR's description as well?

Reply:
Yes, all I meant is some clarification saying something like:

> Note that this PR only handles the case of ... The case of dealing with ... is handled in another PR